mirrord-layer/src/lib.rs

```diff
@@ -252,7 +515,44 @@ async fn poll_agent(mut pf: Portforwarder, mut receiver: Receiver<HookMessage>)
     }
 }
 
+/// Enables file and socket hooks.
 fn enable_hooks() {
-    let interceptor = Interceptor::obtain(&GUM);
-    sockets::enable_hooks(interceptor)
+    let mut interceptor = Interceptor::obtain(&GUM);
+    interceptor.begin_transaction();
+
+    hook!(interceptor, "close", close_detour);
+
+    sockets::enable_socket_hooks(&mut interceptor);
+
+    if env::var("MIRRORD_FILE_OPS").is_ok() {
```

Comment:
we should verify it's true, not just exist

Reply:
Changed it to be a `bool` in the env variable, please check that I added it to the appropriate places.